### ISSUE TITLE
Add regulerization

### ISSUE THREAD

**Issue by EdvinasGervelis** [akemeris/java-neural-net#5]
As a developer, I want to have the ability to disconnect random neurons during network training in order to avoid network overfitting.

In scope:
Overload HiddenLeyer constructor with dropout parameter which takes random neurons disconnection probability.

More information about dropouts can be found here: https://chatbotslife.com/regularization-in-deep-learning-f649a45d6e0
